py3status/modules/google_calendar.py

```diff
@@ -0,0 +1,438 @@
+from oauth2client import tools
+from oauth2client.file import Storage
+from httplib2 import ServerNotFoundError
+
+
+SCOPES = 'https://www.googleapis.com/auth/calendar.readonly'
+APPLICATION_NAME = 'py3status google_calendar module'
+
+
+def gstr_to_datetime(date_time_str):
+    """ Returns a datetime object from a google calendar date/time string."""
+    tmp = '-'.join(date_time_str.split('-')[:-1])
+    return datetime.datetime.strptime(tmp, '%Y-%m-%dT%H:%M:%S')
+
+
+def datetime_to_str(date_time, dt_format):
+    """ Returns a strftime formatted string from a datetime object."""
+    return date_time.strftime(dt_format)
+
+
+def delta_time(date_time):
+    """
+    Returns in a dict the number of days/hours/minutes and total minutes
+    until date_time.
+    """
+    now = datetime.datetime.now()
+    diff = date_time - now
+
+    days = int(diff.days)
+    hours = int(diff.seconds / 3600)
+    minutes = int((diff.seconds / 60) - (hours * 60)) + 1
+    total_minutes = int((diff.seconds / 60) + (days * 24 * 60)) + 1
+
+    return {'days': days,
+            'hours': hours,
+            'minutes': minutes,
+            'total_minutes': total_minutes}
+
+
+class Py3status:
+    auth_token_path = '~/.credentials'
+    button_open_url = 3
+    button_toggle_format = 1
+    cache_timeout = 60
+    client_secret_path = ''
+    colors = []
+    event_in_progress_suffix = 'Remaining'
+    event_separator = '|'
+    format = '{events}'
+    format_date = '%a %d-%m'
+    format_error = ''
+    format_event_compact = '{summary} {time_until}'
+    format_event_full = '{summary} ({start_time} - {end_time}, {start_date})'
+    format_time = '%I:%M %p'
+    format_time_until = '({days}d {hours}h {mins}m)'
+    num_events = 3
+    time_until_threshold = 180
+    warn_threshold = 0
+    warn_timeout = 300
+
+    def post_config_hook(self):
+        self.button = None
+        self.button_states = [False] * self.num_events
+        self.events = None
+        self.no_update = False
+        self.colors.extend(['#FFFFFF'] * (self.num_events - len(self.colors)))
+
+        self.credentials = self._get_credentials()
+        self.is_authorized = self._authorize_credentials()
+
+    def _get_credentials(self):
+        """
+        Gets valid user credentials from storage.
+
+        If nothing has been stored, or if the stored credentials are invalid,
+        the OAuth2 flow is completed to obtain the new credentials.
+
+        Returns: Credentials, the obtained credential.
+        """
+        CLIENT_SECRET_FILE = os.path.join(os.path.expanduser(self.client_secret_path),
+                                          'client_secret.json')
+
+        if not os.path.exists(os.path.expanduser(self.auth_token_path)):
+            os.makedirs(os.path.expanduser(self.auth_token_path))
+
+        self.auth_token_path = os.path.join(os.path.expanduser(self.auth_token_path),
+                                            'py3status-gcal.json')
```

Comment:
`py3status-google-calendar.json`. This is not a `gcal` module. 🤕

Reply:
Changed to `gcal_auth_token.json`.

Reply:
This is `gcal`. https://github.com/ultrabug/py3status/pull/961 It can confuse the users.

Reply:
Crap didn't see this, sorry. I thought you meant the token isn't a module. It's funny, I originally called this module gcal but decided to change it to make it less ambiguous. Then I see you made your own Google Calendar module! Interesting you went with a non-API approach. Simpler in some ways for sure. I'll use a different name for the default token.